spec/features/user_show_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+RSpec.describe 'User show page', type: :feature do
+  let(:user) { FactoryBot.create(:user) }
+  let(:posts) { FactoryBot.create_list(:post, 5, author: user) }
+
+  before do
+    posts
+    visit user_path(user)
+  end
+
+  describe 'User show page' do
+    it 'displays user photo' do
+      expect(page).to have_css("img[alt='User Photo']")
+    end
+
+    it 'displays user bio' do
+      expect(page).to have_content(user.bio)
+    end
+
+    it 'displays user name' do
+      expect(page).to have_content(user.name)
+    end
+
+    it 'displays user posts count' do
+      expect(page).to have_content(user.posts_counter)
+    end
+
+    it 'redirects to post show page' do
+      find_all('a', text: posts.last.title)[0].click
+      expect(page).to have_current_path(user_post_path(user, posts.last))
+    end
+
+    it 'displays user three posts' do
+      expect(page).to have_css('div#limited-posts', count: 3)
+    end
```

Comment:
- [ ] @rmiyoyo and @NoelLincoln Great work adding tests for the users' show page, however this particular test is supposed to test if the `recent_posts` user model method is working correctly, kindly consider reconfiguring the test.

Reply:
I think it is not clear what is required here. Since in the project requirements we are told:
 "I can see the user's first 3 posts." (from my understanding , this means from id 1 up to 3).
 
 But the recent_posts method orders the posts by the most recent going backwards and limits them to three.
 
 Also, the logic I have used in this case is that if the three posts are fetched from the database, ther will be three divs with the id "limited-posts".

Please clarify this , cheers!

Reply:
well resolved! ✔️